### PR TITLE
MAYA-114711 As a user, when clicking on 'Edit as Maya Data' I'd like …

### PR DESCRIPTION
**Pull request by boudrey**
…to get an edit on my sessionLayer instead of my editTarget layer

### CODE REVIEW

File: test/lib/mayaUsd/fileio/testEditAsMaya.py [hunk 1]
```diff
@@ -135,5 +135,38 @@ def testIllegalEditAsMaya(self):
...
+
+        kPullPrimMetadataKey = "Maya:Pull:DagPath"
+        with Usd.EditContext(stage, stage.GetSessionLayer()):
+            self.assertEqual(prim.GetCustomDataByKey(kPullPrimMetadataKey), "|__mayaUsd__|AParent|A")
+
+        # Discard Maya edits, but there is nothing to discard.
+        self.assertTrue(mayaUsd.lib.PrimUpdaterManager.discardEdits("A"))
+
+        # Now Session Layer should be empty, but it is not, it has an empty content
+        self.assertFalse(stage.GetSessionLayer().empty)
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/fileio/testEditAsMaya.py:166]: What is this empty content?

**Review by ppt-adsk — CHANGES_REQUESTED** [review]

File: test/lib/mayaUsd/fileio/testEditAsMaya.py (hunk 1 above)
Comment (boudrey) [reply test/lib/mayaUsd/fileio/testEditAsMaya.py:166]: // USD Layer identifier: anon:00000217FCAC6AE0:anonymousLayer1-session.usda
// Real Path: 
// #usda 1.0
// 
// over "A"
// {
// }
//

Comment (ppt-adsk) [reply test/lib/mayaUsd/fileio/testEditAsMaya.py:165]: O.K., this is what I suspected.  We should be doing this:
https://graphics.pixar.com/usd/release/api/class_sdf_layer.html#a55468b05043d021961a36fdf4912863d
to remove the inert primSpec.  I feel this is out of scope for your pull request, so I'll enter a separate JIRA ticket for it

**Review by ppt-adsk — APPROVED** [review]

File: test/lib/mayaUsd/fileio/testEditAsMaya.py (hunk 1 above)
Comment (ppt-adsk) [reply test/lib/mayaUsd/fileio/testEditAsMaya.py:165]: Entered as MAYA-120677.

**Pull request merged**